skins/10h16/js/lib/reducers/input_validation.js

```diff
@@ -11,11 +11,44 @@ function inputIsValid( value, pattern ) {
 	return new RegExp( pattern ).test( value );
 }
 
+/**
+ * @param {boolean} formWasFilledBefore
+ * @return {Function}
+ */
+function newDataEnteredState( formWasFilledBefore ) {
+	if ( formWasFilledBefore ) {
+		return function () {
+			return true;
+		};
+	} else {
+		return function ( state ) {
+			return state;
+		};
+	}
+}
+
 function inputValidation( validationState, action ) {
 	var newValidationState = objectAssign( {}, validationState ),
-		bankDataIsValid;
+		bankDataIsValid, dataEnteredTransformer;
 
 	switch ( action.type ) {
+		case 'INITIALIZE_VALIDATION':
+			// We have no indicator if the form was freshly loaded with default values, called with payment data from
+			// a banner or reloaded with validation errors. So we try to determine the state of "dataEntered" by
+			// looking at the validation info from the server. In case of coming from the banner,
+			// initialValidationResult is filled, in case of validation errors, violatedFields is filled.
+			// In all other cases, we can't decide and just pass the initial state from the reducer, but that should be ok
+			dataEnteredTransformer = newDataEnteredState(
+				!_.isEmpty( action.payload.violatedFields ) ||
+				!_.isEmpty( action.payload.initialValidationResult )
+			);
+			_.each( validationState, function ( value, key ) {
+				newValidationState[ key ] = {
+					dataEntered: dataEnteredTransformer( validationState[ key ].dataEntered ),
+					isValid: _.has( action.payload.violatedFields, key ) ? true : validationState[ key ].isValid
```

Comment:
Should this not rather be

    isValid: _.has( action.payload.violatedFields, key ) ? false : validationState[ key ].isValid

(a field is not valid if it is contained in violatedFields)?